proto/proto-backplane-grpc/src/main/proto/io/deephaven/proto/table.proto

```diff
@@ -271,11 +273,29 @@ message JoinTablesRequest {
   enum Type {
     CROSS_JOIN = 0;
     NATURAL_JOIN = 1;
-    AS_OF_JOIN = 2;
-    REVERSE_AS_OF_JOIN = 3;
-    EXACT_JOIN = 4;
-    LEFT_JOIN = 5;
+    EXACT_JOIN = 2;
+    LEFT_JOIN = 3;
+  }
+}
+
+message AsOfJoinTablesRequest {
+  enum Type {
+    AS_OF_JOIN = 0;
+    REVERSE_AS_OF_JOIN = 1;
+  }
+  enum MatchRule {
+    LESS_THAN_EQUAL = 0;
+    LESS_THAN = 1;
+    GREATER_THAN_EQUAL = 2;
+    GREATER_THAN = 3;
   }
+  Ticket result_id = 1;
+  TableReference left_id = 2;
+  TableReference right_id = 3;
+  repeated string columns_to_match = 4;
+  repeated string columns_to_add = 5;
+  Type as_of_join_type = 6;
+  MatchRule as_of_match_rule = 7;
```

Comment:
I didn't see anywhere in the code, but do you infer the default values for this if it's null? (Based on the `Type`.)